core/src/main/java/org/apache/carbondata/core/cache/CarbonLRUCache.java

```diff
@@ -43,10 +45,11 @@
   private static final Logger LOGGER =
       LogServiceFactory.getLogService(CarbonLRUCache.class.getName());
   /**
-   * Map that will contain key as table unique name and value as cache Holder
+   * Guava cache that holds key as table unique name and value as cache Holder
    * object
    */
-  private Map<String, Cacheable> lruCacheMap;
+  private Cache<String, Cacheable> lruCacheMap;
```

Comment:
There are scenarios where cache of some table need to be maintained and other tables can be dropped. **It will be good if we support table level expiry.**  and not clear all the cache when expire.

Drop meta cache DML is already supporting table level dropping , but it is manual work now. If this can be handled by guava, it is great

Reply:
Currently, using guava cache, we can do only time-based or size-based eviction for all values(not specific to table level) loaded to CarbonLruCache.

Reply:
@jackylk , @ravipesala : what do you think ?

@Indhumathi27 : I still think we must have **entry level expiry**, just removing all the cache after certain time is not so useful as cache need to be loaded again ?

check this guava issue for the same ([here](https://github.com/google/guava/issues/1203))
I just checked on internet and I found 2 alternatives (under apache license) that supports entry level expiry.
a. [caffeine](https://github.com/ben-manes/caffeine) -- This is on top of guava as they support guava adaptor
b. [expiringmap](https://github.com/jhalterman/expiringmap) -- This is widely used my many companies in their project. see [here](https://github.com/jhalterman/expiringmap/wiki/Who's-Using-ExpiringMap). I guess we can use this one.

Reply:
@Indhumathi27 i agree with @ajantha-bhat , may we need to keep option, or give choices to user and we should have implementations based on that like time based, size based.

Reply:
@ajantha-bhat @akashrn5 ok. I will check those caching libraries and will raise new PR.